### ISSUE TITLE
Control of Ventilation-Device

### ISSUE THREAD

**Issue by silentsimmal** [oysteing/comfoconnect-mqtt-bridge#5]
HI,
I really like the idea, but is there any plan to offer control over mqtt?
[https://github.com/michaelarnauts/aiocomfoconnect](url) has implemented functions to control the device, so I think in combination with a mqtt-subscription it would be a remarkable extension of this project.

BR,
Simon

**Comment by oysteing** [comment]
Hi,

as you point out, device control is an obvious extension, as it's already in the underlying aiocomfoconnect library. Personally, I haven't implemented any control use cases yet.

Do you have any specific use cases in mind, giving an idea what to implement first? Are you interested in using such an extension, and contribute with testing and/or development?

**Comment by silentsimmal** [comment]
In my case, the most helpful control would be ventilation speed and switching between manual and auto-mode.
Further controls like party timer, away mode, bypass On-Off, temperture profile and thresholds, etc. are nice to have, but not the most important ones (for me).
Of course I'm interested in using those extensions and I of course could test them.
If wanted I also could participate in development, but I've got quite limited spare time (I guess like anyone ;) )

**Comment by oysteing** [comment]
Sounds sensible. I've made a first shot at this in #7. You can try it out if you like to. It should be mostly complete. What's not added is the ability to set timeouts for bypass, balancemode, boost, away, comfocoolmode, and temperatureprofile.

**Comment by silentsimmal** [comment]
Thank you for the quick implementation. For your information: I'm using it as docker-image.
A short test of setting and getting mode and speed works great, but only after modifying the reporting-interval parameters.
It seems that they are interpreted as strings, so I had to explicitly cast them as float-values (I locally did this in the __init__-method of the Reporting-class.
I guess there's a better way than my quick and dirty hack, but for the moment it works for me ;)

**Comment by oysteing** [comment]
Oh, the numbers were not cast properly. Is done now. I've casted the seconds as int and the other number as float.

I've merged everything into master.

**Comment by oysteing** [comment]
@silentsimmal FYI, I've made a release with a prebuilt docker image: https://hub.docker.com/r/ogisnas/comfoconnect-mqtt-bridge

**Comment by silentsimmal** [comment]
Ok, good to know.
I planning to extend the methods to set the timeouts manually instead of using the default values of the underlying aiocomfoconnect lib, but as already mentioned my spare time is limited, so it'll take some time.
I'll let you know when I've implemented this.
